test/controllers/email_controller_test.exs

```diff
@@ -0,0 +1,23 @@
+defmodule HexWeb.EmailControllerTest do
+  use HexWeb.ConnCase, async: true
+
+  setup do
+    %{user: create_user("eric", "eric@mail.com", "hunter42")}
+  end
+
+  test "verify email with invalid key", c do
+    email = hd(c.user.emails)
+    conn = get(build_conn(), "email/verify", %{username: c.user.username, email: email.email, key: "invalid"})
+
+    assert redirected_to(conn) == "/"
+    assert get_flash(conn, :error) =~ "failed to verify"
+  end
+
+  test "verify email with valid key", c do
+    email = hd(c.user.emails)
+    conn = get(build_conn(), "email/verify", %{username: c.user.username, email: email.email, key: email.verification_key})
+
+    assert redirected_to(conn) == "/"
+    assert get_flash(conn, :info) =~ "has been verified"
```

Comment:
I came across this when testing user creation but I think doing it in all cases would be fine too. Note sure about security best practices for this kind of thing, but it would be convenient